modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/sorting/Sorting.tsx

```diff
@@ -157,19 +157,68 @@ const AddFDSSortModalContent = ({
 			method: 'POST',
 		});
 
-		if (!response.ok) {
+		if (!addFDSSortResponse.ok) {
 			setSaveButtonDisabled(false);
 
 			openDefaultFailureToast();
 
 			return;
 		}
 
-		const responseJSON = await response.json();
+		const newFDSSort = await addFDSSortResponse.json();
+
+		// If an existing default sorting exists, change its default to false.
+
+		const defaultFDSSort = fdsSorts.find(
+			(fdsSort) => fdsSort.default
+		) as ISort;
+
+		if (useAsDefaultSorting && defaultFDSSort) {
+			const updateDefaultFDSSortResponse = await fetch(
+				`${API_URL.SORTS}/by-external-reference-code/${defaultFDSSort.externalReferenceCode}`,
+				{
+					body: JSON.stringify({
+						default: false,
+					}),
+					headers: {
+						'Accept': 'application/json',
+						'Content-Type': 'application/json',
+					},
+					method: 'PATCH',
+				}
+			);
 
-		openDefaultSuccessToast();
+			if (!updateDefaultFDSSortResponse.ok) {
+				openToast({
+					message: sub(
+						Liferay.Language.get(
+							'an-unexpected-error-occurred-and-the-existing-default-sorting-was-not-removed.please-edit-the-x-sorting-and-uncheck-the-default-option'
+						),
+						[
+							defaultFDSSort.label ||
+								defaultFDSSort.label_i18n[
+									Liferay.ThemeDisplay.getDefaultLanguageId()
+								] ||
+								'',
+						]
+					),
+					type: 'danger',
+				});
+			}
 
-		onSave(responseJSON);
+			const previousDefaultFDSSort =
+				await updateDefaultFDSSortResponse.json();
+
+			onSave({newFDSSort, previousDefaultFDSSort});
+		}
+		else {
+			onSave({newFDSSort});
+		}
+
+		openToast({
+			message: Liferay.Language.get('sorting-was-successfully-added'),
+			type: 'success',
+		});
```

Comment:
Two toast might be a product question, but I think it looks like a bug. In my opinion, we should have only one toast, that is green only after both requests are successful.

Reply:
Ya I think 1 warning toast is probably the most intuitive, with like a "Sorting was successfully added, but now you have multiple defaults when only 1 is allowed". But hopefully this should all be avoided with a single resource action.